internal/controllers/machineset/machineset_controller.go

```diff
@@ -529,45 +614,58 @@ func (r *Reconciler) syncReplicas(ctx context.Context, ms *clusterv1.MachineSet,
-// to be created by the API server, we set the generateName field.
-func (r *Reconciler) getNewMachine(machineSet *clusterv1.MachineSet) *clusterv1.Machine {
-	gv := clusterv1.GroupVersion
-	machine := &clusterv1.Machine{
-		ObjectMeta: metav1.ObjectMeta{
-			GenerateName: fmt.Sprintf("%s-", machineSet.Name),
-			// Note: by setting the ownerRef on creation we signal to the Machine controller that this is not a stand-alone Machine.
-			OwnerReferences: []metav1.OwnerReference{*metav1.NewControllerRef(machineSet, machineSetKind)},
-			Namespace:       machineSet.Namespace,
-			Labels:          make(map[string]string),
-			Annotations:     machineSet.Spec.Template.Annotations,
-		},
-		TypeMeta: metav1.TypeMeta{
-			Kind:       gv.WithKind("Machine").Kind,
-			APIVersion: gv.String(),
-		},
-		Spec: machineSet.Spec.Template.Spec,
-	}
-	machine.Spec.ClusterName = machineSet.Spec.ClusterName
-
-	// Set the labels from machineSet.Spec.Template.Labels as labels for the new Machine.
-	// Note: We can't just set `machineSet.Spec.Template.Labels` directly and thus "share" the labels
-	// map between Machine and machineSet.Spec.Template.Labels. This would mean that adding the
-	// MachineSetNameLabel and MachineDeploymentNameLabel later on the Machine would also add the labels
-	// to machineSet.Spec.Template.Labels and thus modify the labels of the MachineSet.
-	for k, v := range machineSet.Spec.Template.Labels {
-		machine.Labels[k] = v
+func (r *Reconciler) computeDesiredMachine(machineSet *clusterv1.MachineSet, existingMachine *clusterv1.Machine) *clusterv1.Machine {
+	desiredMachine := &clusterv1.Machine{}
+
+	desiredMachine.SetGroupVersionKind(clusterv1.GroupVersion.WithKind("Machine"))
+	desiredMachine.SetNamespace(machineSet.Namespace)
+	desiredMachine.Spec = machineSet.Spec.Template.Spec
+
+	// Set ClusterName
+	desiredMachine.Spec.ClusterName = machineSet.Spec.ClusterName
+
+	// Set OwnerReferences
+	desiredMachine.OwnerReferences = util.EnsureOwnerRef(
+		desiredMachine.OwnerReferences,
+		*metav1.NewControllerRef(machineSet, machineSetKind),
+	)
+
+	// Set Labels
+	desiredMachine.Labels = machineLabels(machineSet)
+
+	// Set Annotations
+	desiredMachine.Annotations = map[string]string{}
+	for k, v := range machineSet.Spec.Template.Annotations {
+		desiredMachine.Annotations[k] = v
+	}
+
+	if existingMachine == nil {
+		// Creating a new Machine
+		desiredMachine.SetName(names.SimpleNameGenerator.GenerateName(fmt.Sprintf("%s-", machineSet.Name)))
+	} else {
+		// Updating an existing Machine
+		desiredMachine.SetName(existingMachine.Name)
+		desiredMachine.SetUID(existingMachine.UID)
+		desiredMachine.Spec.InfrastructureRef = existingMachine.Spec.InfrastructureRef
+		desiredMachine.Spec.Bootstrap.ConfigRef = existingMachine.Spec.Bootstrap.ConfigRef
 	}
 
+	return desiredMachine
+}
+
+func machineLabels(ms *clusterv1.MachineSet) map[string]string {
+	machinelabels := map[string]string{}
```

Comment:
```suggestion
	machineLabels := map[string]string{}
```
nit: just because I saw it in Intellij